src/ValueObject/Type/AbstractComposite.php

```diff
@@ -13,9 +13,22 @@ public function equals(ValueObject $other_valueobject)
             return $result;
         }
         foreach ($this as $key=>$valueobject) {
-            $result = $result && 
-                $valueobject->equals($other_valueobject->$key);
+
+            if ($valueobject === null) {
+                $result = $result && ($other_valueobject->$key === null);
+            } else {
+                $result = $result && $valueobject->equals($other_valueobject->$key);
+            }
         }
 		return $result;
 	}
+
+    public function is_null()
+    {
+        $result = true;
+        foreach ($this as $key=>$valueobject) {
+            $result = $result && ($valueobject === null);
```

Comment:
why not if ($valueobject !== null) return false; and then outside of the foreach just a return true ?

would skip a few iterations in the foreach if it finds at least one object !null